test/suites/standard/networking-smoke.robot

```diff
@@ -63,6 +65,20 @@ Delete Hello MicroShift Route
     [Documentation]    Delete route for cleanup.
     Oc Delete    route/hello-microshift -n ${NAMESPACE}
 
+Wait For Service Deletion With Timeout
```

Comment:
Is `--wait=true` not waiting for complete deletion in the delete keyword above?

Reply:
Do you mean on line 66? I would expect a wait there to actually block, and the controllers to delete the service and endpoint before deleting the route.

Maybe the `Oc Delete` keyword should always add `--wait`?

Reply:
I'm going to 2x this. Before I went this route, I'd tried `--wait` and examined to logs to see what the deletion sequence looked like.  AFAICT though, the deletions are asynchronous.  This makes sense as the service does not have a finalizer placed on it to pause it's deletion.

Reply:
Examining the logs + referring to upstream docs, the EP deletion is asynchronous.  The endpoint is not reconciled until the service disappears:
```
Dec 04 12:10:38 el92-src-standard-suite-host1 microshift[13935]: microshift-loadbalancer-service-controller I1204 12:10:38.429501   13935 controller.go:150] Service default/hello-microshift does not exist anymore
Dec 04 12:10:38 el92-src-standard-suite-host1 microshift[13935]: kube-controller-manager I1204 12:10:38.435518   13935 garbagecollector.go:672] "Deleting item" item="[discovery.k8s.io/v1/EndpointSlice, namespace: default, name: hello-microshift-b7jvm, uid: ce614dbc-e818-47cd-a880-62b8778a5b25]" propagationPolicy="Background"
```

The delta between the events on the backend is on the order of hundredths of a second, but that does not include the latency in updating the cluster's object store to the latest cluster state.

Reply:
OK, it sounds like we need the test to be a little smart about what it waits for.